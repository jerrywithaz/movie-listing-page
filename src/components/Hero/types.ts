export type HeroProps = React.HTMLAttributes<HTMLDivElement>;