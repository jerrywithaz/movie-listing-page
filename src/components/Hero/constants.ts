export const SLANT_ANGLE_DEGREES = Math.atan(80/1440) * 180/Math.PI;
export const SLANT_ANGLE_RADIANS = Math.atan(80/1440) * 180/Math.PI;