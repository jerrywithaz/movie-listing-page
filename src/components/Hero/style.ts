import styled from 'styled-components';

export const Hero = styled.div`
    background: linear-gradient(
        159.41deg, 
        ${props => props.theme.palette.secondary} 3.42%, 
        ${props => props.theme.palette.tertiary} 96.5%
    );
    height: 560px;
    width: 100%;
    position: relative;
    overflow: hidden;
    z-index: 1;
    display: flex;
    align-items: center;
    justify-content: stretch;
    flex-direction: column;
    &:before {
        background: ${props => props.theme.palette.gray.primary};
        bottom: -25%;
        content: '';
        display: block;
        height: 50%;
        left: 0;
        position: absolute;
        right: 0;
        transform: skewY(-3deg);
        transform-origin: 100%;
        z-index: -1;
    }
`;

export const HeroInner = styled.div`
    position: relative;
`;

export const Arrow = styled.img`
    width: 260px;
    height: auto;
    display: block;
    margin: 0 auto;
`;

export const Logo = styled.img`
    width: 552px;
    height: auto;
    display: block;
    margin: 0 auto;
`;