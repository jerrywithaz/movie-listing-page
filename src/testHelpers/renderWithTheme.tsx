import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;