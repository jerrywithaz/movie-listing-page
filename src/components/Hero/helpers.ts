import { SLANT_ANGLE_RADIANS } from './constants';


export function createSlant() {
    
    const adjacent = window.innerWidth;
    const hypotenuse = Math.abs(adjacent / Math.cos(SLANT_ANGLE_RADIANS));
    const opposite = Math.abs(Math.sin(SLANT_ANGLE_RADIANS) * hypotenuse);
    const angle = Math.atan(opposite/adjacent) * 180/Math.PI;

    return {adjacent, angle, hypotenuse, opposite};

}