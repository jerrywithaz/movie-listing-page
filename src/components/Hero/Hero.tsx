import React from 'react';
import arrow from 'assets/images/hero-arrow.svg';
import logo from 'assets/images/hero-logo.svg';

import * as Styled from './style';

const Hero = () => {

    return (
        <Styled.Hero>
            <Styled.HeroInner>
                <Styled.Arrow src={arrow} alt="Welcome to"/>
                <Styled.Logo src={logo} alt="Alamo Drafthouse Cinema"/>
            </Styled.HeroInner>
        </Styled.Hero>
    );

};

export default Hero;