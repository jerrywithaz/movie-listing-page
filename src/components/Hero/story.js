import React from 'react';
import { storiesOf } from '@storybook/react';
import Hero from './';

const story = storiesOf('Hero', module);

story.add('Default', () => <Hero/>);
  