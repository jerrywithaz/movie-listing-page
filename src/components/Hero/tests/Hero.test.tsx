import React from 'react';
import Hero from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';

describe("components/Hero", () => {

    it("should match snapshot", () => {

        const tree = renderWithTheme(<Hero/>).toJSON();

        expect(tree).toMatchSnapshot();

    });

});