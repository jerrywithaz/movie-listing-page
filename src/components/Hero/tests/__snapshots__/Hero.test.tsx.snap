// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Hero should match snapshot 1`] = `
.c0 {
  background: linear-gradient( 159.41deg, #FF009D 3.42%, #F5B324 96.5% );
  height: 61.11111111111111%;
  width: 100%;
  position: relative;
  overflow: hidden;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:before {
  background: #E5E5E5;
  bottom: -25%;
  content: '';
  display: block;
  height: 50%;
  left: 0;
  position: absolute;
  right: 0;
  -webkit-transform: skewY(-3deg);
  -ms-transform: skewY(-3deg);
  transform: skewY(-3deg);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: -1;
}

.c1 {
  position: relative;
}

.c2 {
  width: 260px;
  height: auto;
  display: block;
  margin: 0 auto;
}

.c3 {
  width: 552px;
  height: auto;
  display: block;
  margin: 0 auto;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Welcome to"
      className="c2"
      src="hero-arrow.svg"
    />
    <img
      alt="Alamo Drafthouse Cinema"
      className="c3"
      src="hero-logo.svg"
    />
  </div>
</div>
`;
